lua/lib/reflection.lua

```diff
@@ -188,8 +190,22 @@ function reflection:begin_playback()
   self.step = 0
   self.play = 1
   self.start_callback()
+  local queued_start_callback = false
+  local queued_end_of_loop_callback = false
+  local queued_end_playback = false
   while self.play == 1 do
-    clock.sync(1 / 96)
+    if queued_start_callback then
+      self:start_callback()
+      queued_start_callback = false
+    end
+    if queued_end_of_loop_callback then
+      self.end_of_loop_callback()
+      queued_end_of_loop_callback = false
+    end
+    if queued_end_playback then
+      self:end_playback()
```

Comment:
interesting: the way this is set up has a new possible edge case where the current step processes one last time _after_ `end_playback()` is called. Maybe better to add a `break` here so that doesn't happen?

Reply:
_good_ **call**